test/unit/router.test.js

```diff
@@ -63,4 +63,14 @@ describe('Router', () => {
       expect(Object.keys(pageLoader.loaded)).toEqual(routes)
     })
   })
+
+  describe('.urlIsNew()', () => {
+    it('should handle undefined starting query parameters', () => {
+      const pageLoader = new PageLoader()
+      const router = new Router('/', undefined, '/', { pageLoader })
```

Comment:
I wonder in what case this happens 🤔 Could you also add an integration test

Reply:
This happens in the case described in https://github.com/zeit/next.js/issues/2943. 

I'm not sure I see the value in a full on integration test. They tend to be hard to run, somewhat flaky and hard to maintain so I've always favored saving them to smoke test critical paths. This was my experience from working on the Walmart team and with the  https://github.com/TestArmada/magellan project anyway. I'm happy to write another unit test if you can think of other ways to cause this behavior. 

If you feel strongly that an integration test is the right move here, can you give me some guidance on running the test suite? It nearly crashed my computer and left a bunch of orphaned Chrome processes around last time. Is there a way to do more tarted runs?